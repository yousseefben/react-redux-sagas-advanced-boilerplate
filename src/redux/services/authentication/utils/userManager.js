import { createUserManager } from 'redux-oidc';

const userManagerConfig = {
  client_id: 'example-app',
  redirect_uri: `${window.location.protocol}//${window.location.hostname}${
    window.location.port ? `:${window.location.port}` : ''
  }/callback`,
  response_type: 'token id_token',
  // scope: 'openid profile https://www.googleapis.com/auth/youtube.readonly',
  authority: 'http://localhost:8080/auth/realms/test',
  silent_redirect_uri: `${window.location.protocol}//${
    window.location.hostname
  }${window.location.port ? `:${window.location.port}` : ''}/silent_renew.html`,
  automaticSilentRenew: true,
  filterProtocolClaims: true,
  loadUserInfo: true
};

const userManager = createUserManager(userManagerConfig);

export default userManager;
