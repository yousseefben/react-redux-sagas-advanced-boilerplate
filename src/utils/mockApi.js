const axios = require('axios');
const MockAdapter = require('axios-mock-adapter');

export const mock = new MockAdapter(axios);
