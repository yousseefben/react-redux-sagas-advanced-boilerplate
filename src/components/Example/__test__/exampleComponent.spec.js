import React from 'react';

import { fireEvent, cleanup, waitForElement } from '@testing-library/react';
import ExampleComponent from '../ExampleComponent';
import example from '../redux/exampleActions';
import { renderWithRedux } from '../../../utils/test_utils';

afterEach(cleanup);

const init = {
  initialState: {
    exampleData: {
      data: [
        {
          id: 1,
          title: 'title'
        }
      ],
      fetching: false
    }
  }
};

test('can dipsatch action and show loading', async () => {
  const { getByText } = renderWithRedux(<ExampleComponent />);

  jest.spyOn(example, 'get');

  fireEvent.click(getByText('Click here!'));
  expect(example.get).toHaveBeenCalledTimes(1);
  await waitForElement(() => expect(getByText('Loading data ...')));
});

test('can render component', async () => {
  const { getByTestId } = renderWithRedux(<ExampleComponent />);

  expect(getByTestId('div')).toMatchSnapshot();
});
