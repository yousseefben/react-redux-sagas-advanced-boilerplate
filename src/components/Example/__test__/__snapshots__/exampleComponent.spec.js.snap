// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render component 1`] = `
<div
  data-testid="div"
>
  hello example!
  <table>
    <tbody>
      <tr>
        <th>
          id
        </th>
        <th>
          title
        </th>
        <th>
          Body
        </th>
      </tr>
    </tbody>
  </table>
  <button
    class="ant-btn"
    type="button"
  >
    <span>
      Click here!
    </span>
  </button>
</div>
`;
